diff --git a/apis/v1/doc.go b/apis/v1/doc.go
--- a/apis/v1/doc.go
+++ b/apis/v1/doc.go
@@ -16,4 +16,5 @@
 // +kubebuilder:object:generate=true
 // +k8s:deepcopy-gen=package,register
 // +groupName=secrets-store.csi.x-k8s.io
+// +k8s:openapi-gen=true
 package v1
diff --git a/apis/v1alpha1/doc.go b/apis/v1alpha1/doc.go
--- a/apis/v1alpha1/doc.go
+++ b/apis/v1alpha1/doc.go
@@ -16,4 +16,5 @@
 // +kubebuilder:object:generate=true
 // +k8s:deepcopy-gen=package,register
 // +groupName=secrets-store.csi.x-k8s.io
+// +k8s:openapi-gen=true
 package v1alpha1
